### PR TITLE
[test] Restrict image coverage for allowlist feature

### PR DESCRIPTION
**Pull request by shantanutrip**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Allowlist feature was getting activated for MxNet1.9. This PR restricts the image coverage for the allowlist feature.

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils/security.py [hunk 1]
```diff
@@ -494,8 +494,11 @@ def conduct_failure_routine(
         "fixable_vulnerabilities": fixable_list,
         "non_fixable_vulnerabilities": newly_found_non_fixable_list,
     }
-    ## TODO: Remove the is_pr_context before merging ##
-    if test_utils.is_canary_context() and test_utils.is_time_for_invoking_ecr_scan_failure_routine_lambda():
+    ## TODO: Make the conditions below as if test_utils.is_canary_context() and test_utils.is_time_for_invoking_ecr_scan_failure_routine_lambda() and os.getenv("REGION") == test_utils.DEFAULT_REGION:
+    ## to make sure that we just invoke the ECR_SCAN_FAILURE_ROUTINE_LAMBDA once everyday
+    if test_utils.is_canary_context() and os.getenv("REGION") == test_utils.DEFAULT_REGION:
```
Comment (shantanutrip) [inline test/test_utils/security.py:499]: I am planning to push it like this to get feedback faster from the canaries. Once the canaries succeed, I will revert this as per the TODO statement I have commented above.

File: test/dlc_tests/sanity/test_security_check.py [hunk 2]
```diff
@@ -58,7 +62,11 @@ def is_image_covered_by_allowlist_feature(image):
     image_framework, image_version = get_framework_and_version_from_tag(image)
     if image_framework not in LOWER_THRESHOLD_IMAGES or any(substring in image for substring in ["example"]):
         return False
-    if Version(image_version) >= Version(LOWER_THRESHOLD_IMAGES[image_framework]):
+    if Version(image_version) >= Version(LOWER_THRESHOLD_IMAGES[image_framework]["minimum_version_included"]):
+        if "maximum_version_excluded" in LOWER_THRESHOLD_IMAGES[image_framework]:
+            if Version(image_version) < Version(LOWER_THRESHOLD_IMAGES[image_framework]["maximum_version_excluded"]):
+                return True
+            return False
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_security_check.py:69]: Please use `packaging.specifiers.SpecifierSet` to define upper and lower limits.

Comment (shantanutrip) [reply test/dlc_tests/sanity/test_security_check.py]: Thanks!! Made the change.

**Review by saimidu — DISMISSED** [review]

File: test/dlc_tests/sanity/test_security_check.py [hunk 3]
```diff
@@ -25,7 +26,7 @@
 )
 from src.config import is_ecr_scan_allowlist_feature_enabled
 
-LOWER_THRESHOLD_IMAGES = {"mxnet": "1.8.0"}
+LOWER_THRESHOLD_IMAGES = {"mxnet":SpecifierSet(">=1.8.0,<1.9.0")}
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_security_check.py:29]: is this global used anywhere outside this file?

Comment (shantanutrip) [reply test/dlc_tests/sanity/test_security_check.py:29]: No, it is not used anywhere outside the file.

File: test/test_utils/security.py (hunk 1 above)
Comment (arjkesh) [reply test/test_utils/security.py:499]: understood - thanks for explanation

File: test/dlc_tests/sanity/test_security_check.py (hunk 3 above)
Comment (arjkesh) [reply test/dlc_tests/sanity/test_security_check.py:29]: nit: this is now lower and upper threshold. do you want to change the name?

Comment (arjkesh) [reply test/dlc_tests/sanity/test_security_check.py:29]: style nit: can you add space after ":"

**Review by saimidu — APPROVED** [review]

**Pull request merged**